hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -328,24 +328,54 @@ public StreamSync(HoodieStreamer.Config cfg, SparkSession sparkSession,
   }
 
   /**
-   * Refresh Timeline.
+   * Creates a meta client for the table, and refresh timeline.
    *
    * @throws IOException in case of any IOException
    */
-  public void refreshTimeline() throws IOException {
+  public HoodieTableMetaClient initializeMetaClientAndRefreshTimeline() throws IOException {
+    return initializeMetaClient(true);
+  }
+
+  /**
+   * Creates a meta client for the table without refreshing timeline.
+   *
+   * @throws IOException in case of any IOException
+   */
+  private HoodieTableMetaClient initializeMetaClient() throws IOException {
+    return initializeMetaClient(false);
+  }
+
+  /**
+   * Creates a meta client for the table, and refresh timeline if specified.
+   * If the table does not yet exist, it will be initialized.
+   *
+   * @param refreshTimeline when set to true, loads the active timeline and updates the {@link #commitsTimelineOpt} and {@link #allCommitsTimelineOpt} values
+   * @return a meta client for the table
+   * @throws IOException in case of any IOException
+   */
+  private HoodieTableMetaClient initializeMetaClient(boolean refreshTimeline) throws IOException {
     if (storage.exists(new StoragePath(cfg.targetBasePath))) {
```

Comment:
This can be further optimized by using try-catch to assume the existence of `targetBasePath` and handle the new table case by catching `TableNotFoundException`, so we don't need to check target path for every sync. But I think we can use a separate PR to fix this